autonomy/deploy/generators/docker_compose/base.py

```diff
@@ -215,13 +215,11 @@ def generate_config_tendermint(self) -> "DockerComposeGenerator":
         if not self.use_tm_testnet_setup:
             return self
 
-        hosts = (
-            " \\\n".join(
-                [
-                    f"--hostname=node{k}"
-                    for k in range(self.service_builder.service.number_of_agents)
-                ]
-            ),
+        hosts = " \\\n".join(
+            [
+                "--hostname=" + self.service_builder.get_tm_container_name(index=k)
+                for k in range(self.service_builder.service.number_of_agents)
+            ]
```

Comment:
Can you explain how this resolves the issue?

Reply:
Before the fix the hostnames were being formatted as `--hostname=node{NODE_ID}`, with this change they'll be formatted as `--hostname={SERVICE_NAME}_tm_{NODE_ID}`

Reply:
I understand this part. I do not understand the situation though. I do not know what issue this PR is trying to solve. The description does not help. Is it simply something that was missed on #2016? What is the symptom of the issue? Why was `hosts` a `tuple` and now it is a string?

Reply:
So, when building deployments that uses pre-established tendermint networks we build a testnet using `tendermint testnet` command and use this a said pre-established tendermint network. Here when defining the number of validators we also define the hostnames for every node using `--hostname=<NAME>`. And this was missed during the last release.